velox/connectors/hive/HiveWriteProtocol.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/WriteProtocol.h"
+
+namespace facebook::velox::connector::hive {
+
+/**
+ * Parameters for Hive writers
+ */
+class HiveWriterParameters : public WriterParameters {
+ public:
+  /// The writing is creating a new directory, appending files to the directory,
+  /// or overwriting a directory
+  enum class UpdateMode { kNew, kAppend, kOverwrite };
+
+  HiveWriterParameters(
```

Comment:
Would you document the arguments and explain the difference between "target" and "write"?

Reply:
Documented for class members